simulation/halsim_ws_core/src/main/native/cpp/WSProvider_Accelerometer.cpp

```diff
@@ -0,0 +1,55 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "WSProvider_Accelerometer.h"
+
+#include <hal/Ports.h>
+#include <hal/simulation/AccelerometerData.h>
+
+#define REGISTER(halsim, jsonid, ctype, haltype)                          \
+  HALSIM_RegisterAccelerometer##halsim##Callback(                         \
+      m_channel,                                                          \
+      [](const char* name, void* param, const struct HAL_Value* value) {  \
+        static_cast<HALSimWSProviderAccelerometer*>(param)                \
+            ->ProcessHalCallback(                                         \
+                {{jsonid, static_cast<ctype>(value->data.v_##haltype)}}); \
+      },                                                                  \
+      this, true)
+namespace wpilibws {
+void HALSimWSProviderAccelerometer::Initialize(WSRegisterFunc webRegisterFunc) {
+  CreateProviders<HALSimWSProviderAccelerometer>("Accelerometer", 1,
+                                                 webRegisterFunc);
+}
+
+HALSimWSProviderAccelerometer::~HALSimWSProviderAccelerometer() {
+  DoCancelCallbacks();
+}
+
+void HALSimWSProviderAccelerometer::RegisterCallbacks() {
+  m_activeCbKey = REGISTER(Active, "<active", bool, boolean);
+  m_rangeCbKey = REGISTER(Range, "<range", int32_t, int);
+  m_xCbKey = REGISTER(X, "<x", int32_t, int);
```

Comment:
The field identifiers for x, y and z should be `>x`, `>y` and `>z` instead (since they are inputs into robot code)